src/NLog/Targets/FileTarget.cs

```diff
@@ -1908,6 +1915,24 @@ private bool PreviousLogOverlappedPeriod(LogEventInfo logEvent, DateTime lastWri
         }
 
         /// <summary>
+        /// Calculate the DateTime of the requested day of the week.
+        /// </summary>
+        /// <param name="previousLogEventTimestamp">The DateTime of the previous log event.</param>
+        /// <param name="dayOfWeek">The next occuring day of the week to return a DateTime for.</param>
+        /// <returns>The DateTime of the next occuring dayOfWeek.</returns>
+        /// <remarks>For example: if previousLogEventTimestamp is Thursday 2017-03-02 and dayOfWeek is Sunday, this will return
+        ///  Sunday 2017-03-05. If dayOfWeek is Thursday, this will return *next* Thursday 2017-03-09.</remarks>
+        public DateTime CalculateNextWeekday(DateTime? previousLogEventTimestamp, DayOfWeek dayOfWeek)
+        {
+            // Shamelessly taken from http://stackoverflow.com/a/7611480/1354930
+            int start = (int)previousLogEventTimestamp.Value.DayOfWeek;
```

Comment:
> How should I adding nullref handling to CalcualteNextWeekday?

Change the parameter from `DateTime?` to `DateTime`. Then it's fixed in this case :)

(I see indeed it's already checked in the calls)